gitnamehere/point-of-sale-group-project
# POS - Implement voiding of orders

**Ticket:** **[ATS-51: POS - Finish implementing the ordering process](https://cs375-pos-group-project.atlassian.net/browse/ATS-51?atlOrigin=eyJpIjoiYjgzYmJhMDljODFiNGZjOWI2MThiODc0ZDY2YmYyNjgiLCJwIjoiaiJ9)](https://cs375-pos-group-project.atlassian.net/browse/ATS-52?atlOrigin=eyJpIjoiYjA2YzRjNmJlNzFmNDcwOGFhY2IyZGY1YzIxMDNkYzMiLCJwIjoiaiJ9)**

- Added API endpoints for voiding and refunding
- Added buttons to payment screen for both voiding and refunding

Testing:
- `npm run setup-demo:dev` (If you have not updated your DB recently)
- `npm run start:dev`
- Place order(s) at `http://localhost:3000/pos/ordering/`
- Pay for order at `http://localhost:3000/pos/payment/?id=1`
- Verify orders which have been paid can be refunded
- Verify orders that have not been paid cannot be refunded (nothing should change)
- Verify orders which have been paid cannot be voided (nothing should change)
- Verify orders which have not been paid can be voided
- Verify the above is reflected in the Orders page (play around with the filters)
- Verify upon refunding an order, it is updated in the DB at `http://localhost:3000/api/orders` and user is redirected to `http://localhost:3000/pos/orders/`
- Verify upon voiding an order, it is updated in the DB at `http://localhost:3000/api/orders` and user is redirected to `http://localhost:3000/pos/orders/`
